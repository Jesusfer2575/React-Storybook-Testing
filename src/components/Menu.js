import React, { useState } from 'react';
import { Dropdown, InputGroup, FormControl } from 'react-bootstrap';
import { typeOperation } from '../utils/constants';
import Calculator from './Calculator';

const Menu = () => {
  const [operation, setOperation] = useState(1);
  const [valueA, setValueA] = useState(0);
  const [valueB, setValueB] = useState(0);
  
  const handleSelect = (eventKey) => {
    setOperation(parseInt(eventKey));
  };
  
  return (
    <div>
      <Dropdown onSelect={handleSelect} id="menuDropdown">
        <Dropdown.Toggle variant="success" id="dropdown-basic">
          Select an operation
        </Dropdown.Toggle>
        <Dropdown.Menu>
          <Dropdown.Item eventKey={typeOperation.SUM}>Addition</Dropdown.Item>
          <Dropdown.Item eventKey={typeOperation.SUBSTRACTION}>Substraction</Dropdown.Item>
          <Dropdown.Item eventKey={typeOperation.MULTIPLICATION}>Multiplication</Dropdown.Item>
          <Dropdown.Item eventKey={typeOperation.DIVISION}>Division</Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
      <InputGroup size="sm" className="mb-3">
        <FormControl aria-label="Small" aria-describedby="inputGroup-sizing-sm" onChange={(e) => setValueA(e.target.value)}/>
        <FormControl aria-label="Small" aria-describedby="inputGroup-sizing-sm" onChange={(e) => setValueB(e.target.value)}/>
      </InputGroup>
      <Calculator valueA={parseInt(valueA)} valueB={parseInt(valueB)} operation={operation}/>
    </div>
      
  );
}
 
export default Menu;