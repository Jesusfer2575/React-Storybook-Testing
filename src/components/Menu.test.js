import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu component tests', () => {
  it('Should change the state when select the dropdown', () => {
    const component = shallow(<Menu />);
    expect(component).toMatchSnapshot();
  });
});