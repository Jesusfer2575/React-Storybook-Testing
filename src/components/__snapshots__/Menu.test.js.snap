// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component tests Should change the state when select the dropdown 1`] = `
<div>
  <Dropdown
    id="menuDropdown"
    navbar={false}
    onSelect={[Function]}
  >
    <DropdownToggle
      id="dropdown-basic"
      variant="success"
    >
      Select an operation
    </DropdownToggle>
    <DropdownMenu
      alignRight={false}
      flip={true}
    >
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        eventKey={1}
      >
        Addition
      </DropdownItem>
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        eventKey={2}
      >
        Substraction
      </DropdownItem>
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        eventKey={3}
      >
        Multiplication
      </DropdownItem>
      <DropdownItem
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        eventKey={4}
      >
        Division
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
  <InputGroup
    className="mb-3"
    size="sm"
  >
    <FormControl
      aria-describedby="inputGroup-sizing-sm"
      aria-label="Small"
      onChange={[Function]}
    />
    <FormControl
      aria-describedby="inputGroup-sizing-sm"
      aria-label="Small"
      onChange={[Function]}
    />
  </InputGroup>
  <Calculator
    operation={1}
    valueA={0}
    valueB={0}
  />
</div>
`;
